lib/mocha/expectation.rb

```diff
@@ -222,6 +223,7 @@ def with(*expected_parameters, &matching_block)
       @parameters_matcher = ParametersMatcher.new(expected_parameters, &matching_block)
```

Comment:
Hmm, since `expected_parameters` can be matchers as well, would that break matchers for "Any" keyword arguments? e.g. `.with(has_value(...))` might be interpreted as a positional argument when it could previously be used to match against keyword args.

Reply:
In the current implementation it still works, this passes:
```ruby
def test_has_value
  created_widget = Widget.new
  Widget.expects(:create).with(has_value('wombat')).returns(created_widget)
  assert_equal created_widget, Widget.create(:model => 'wombat')
end
```
However, I think it's because the keyword arg check only happens if the last param is a Hash, which in this case it isn't:
https://github.com/freerange/mocha/blob/44e7d4161678fcdc766bb36210611058f9298d6f/lib/mocha/parameters_matcher.rb#L32
This might become problem when we want to refactor parameters in the future (see [comment](https://github.com/freerange/mocha/pull/534#discussion_r956200705)).

Reply:
To discuss further in https://github.com/freerange/mocha/pull/534#discussion_r959743155